### ISSUE TITLE
Redirect errors in Google Search Console Report

### ISSUE THREAD

**Issue by tpmccallum** [fermyon/developer#451]
The following URLs are showing redirect errors associated with them:
- https://developer.fermyon.com/spin/quickstart/rust-components
- https://developer.fermyon.com/spin/quickstart/go-components
- https://developer.fermyon.com/spin/quickstart/configuration
- https://developer.fermyon.com/spin/configuration/redis-trigger
- https://developer.fermyon.com/spin/rust-components/redis-trigger

We most likely need to:
- look at the `developer/spin.toml` file and add/remove/update redirects
- ensure the [sitemap.xml](https://developer.fermyon.com/sitemap.xml) is showing these pages correctly

**Comment by tpmccallum** [comment]
Also is it possible to either block (via the `robots.txt` file) or redirect via the `spin.toml` file; the following URLs (which are showing as 404 Not Found errors):
- https://developer.fermyon.com/spin/go-components/http-trigger
- https://developer.fermyon.com/spin/qu...
- https://developer.fermyon.com/spin/go-components/redis-trigger
- https://developer.fermyon.com/events/2023-developerweek-sf
- https://developer.fermyon.com/spin/configuration/developing
- https://developer.fermyon.com/spin/configuration/dynamic-configuration
- https://developer.fermyon.com/cloud/contributing

The one with the `...` (https://developer.fermyon.com/spin/qu...) is a link in an external YouTube video description that was automatically shortened in that YouTube video's description (we have no control over that sort of external 3rd-party URL issue that points to our base domain; so hoping there is something we can do internally to deal with these sorts of things by hand when/if they appear).

These are a bit tricky, but there is only a handful, and it would be great to resolve them in the near future if possible.

**Comment by tpmccallum** [comment]
Let's look at the first URL at the top of this issue page/description to elaborate.

For the https://developer.fermyon.com/spin/quickstart/rust-components URL, there is a redirect to the newer
https://developer.fermyon.com/spin/rust-components URL driven by the `spin.toml` file's redirect on line 192
https://github.com/fermyon/developer/blob/main/spin.toml#L192

In this case, it appears that we can deem this as a replacement page, and therefore, it might be worth implementing a permanent redirect `301` because it looks like the redirect is a temporary `302` i.e.

```bash
curl -vi https://developer.fermyon.com/spin/quickstart/rust-components
> GET /spin/quickstart/rust-components HTTP/2
> Host: developer.fermyon.com
< HTTP/2 302 
HTTP/2 302 
< location: https://developer.fermyon.com/spin/rust-components
location: https://developer.fermyon.com/spin/rust-components
```

Actually it appears that this may also be the solution for the next URL i.e. use `301` instead of `302` and all of the others in that first set of URLs **[1]**.

```bash
curl -vi https://developer.fermyon.com/spin/quickstart/go-components  
> GET /spin/quickstart/go-components HTTP/2
> Host: developer.fermyon.com
< HTTP/2 302 
HTTP/2 302 
< location: https://developer.fermyon.com/spin/go-components
location: https://developer.fermyon.com/spin/go-components
```


**[1]** 
https://developer.fermyon.com/spin/quickstart/rust-components
https://developer.fermyon.com/spin/quickstart/go-components
https://developer.fermyon.com/spin/quickstart/configuration
https://developer.fermyon.com/spin/configuration/redis-trigger
https://developer.fermyon.com/spin/rust-components/redis-trigger

**Comment by itowlson** [comment]
This would require a (small) amount of work on the redirector component, which is currently hardwired to status code 302.

**Comment by tpmccallum** [comment]
Also, just inspecting the rest of those URLs (in the [second comment block, of this GH issue](https://github.com/fermyon/developer/issues/451#issuecomment-1465320841)), which are all returning `404` errors, we could also implement `301`s if we knew 100% where those URL were intending to resolve.

As far as the external URLs like the shortened ones from third-party sites that end in `...` the user will logically receive the following page (which in fairness does have guidance to find home or spin docs or cloud docs etc). Perhaps we could create a new GH task to make the `404` more visually appealing than just small text with links @flynnduism, any ideas on giving the `404` a makeover?

 
![Screenshot 2023-03-13 at 11 09 06 am](https://user-images.githubusercontent.com/9831342/224586833-6bd61dfb-e8b3-4db8-ba99-fc7b1e0289bf.png)

**Comment by tpmccallum** [comment]
Thanks @itowlson 
This has been an interesting dive into our URLs; perhaps we can close this issue and open a new one just covering the `302` to `301` work and a separate issue that covers working on giving the `404` page an upgrade. Does anyone agree; any extra comments?
(Don't want one issue to split into multiple tasks/conversations) :)

**Comment by itowlson** [comment]
Yep, we already have an issue for the 301 work: https://github.com/fermyon/spin-redirect/issues/3

**Comment by quyumkehinde** [comment]
This seems to be done.
